app/generators/days_generator.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+class DaysGenerator < QuestionTypeGenerator
+  DATE_FORMAT = '%A %e %B'
+
+  def generate(question)
+    title = safe_join([question[:title].html_safe, generate_tooltip(question[:tooltip])])
+    checkbox_group = safe_join([
+                                 tag.p(title, class: 'flow-text'),
+                                 answer_options(question)
+                               ])
+    tag.div(checkbox_group, class: checkbox_group_klasses(question))
+  end
+
+  private
+
+  def checkbox_group_klasses(question)
+    klasses = 'checkbox-group'
+    klasses += ' required' if question[:required].present?
+    klasses
+  end
+
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/PerceivedComplexity
+  # rubocop:disable Metrics/MethodLength
+  def answer_options(question)
+    body = []
+    today = question[:include_today].present? ? 0 : 1
+    from_days_ago = question[:from_days_ago]
+    date_format = question[:date_format].presence || DATE_FORMAT
+    from_days_ago.downto(today) do |days_ago|
+      date = days_ago.days.ago
+      next if [6, 0].include?(date.wday) && question[:exclude_weekends].present?
+
+      formatted_date = I18n.l(date, locale: question[:locale], format: date_format)
+      option = {
+        title: formatted_date,
+        value: I18n.l(date, locale: :en, format: '%Y-%m-%d')
```

Comment:
why don't you want to use `date_format` here? Should this special format be in a constant?

Reply:
Good point. Let me fix it.

Reply:
Sorry, now I remember again. This is the value that is used as form value, i.e., the user never sees this value. It is merely what will be sent to the backend. And for this I wanted something that was 1) obviously always the same, no matter the user's locale (that's why it is always in English), and 2) something that would be easy to parse, but was still custom made, so it's the date, optionally followed by AM or PM.

I'll move everything to constants and add an explanation.

Reply:
Fixed.